assets/tests/components/map/utilities/reactDivIcon.test.tsx

```diff
@@ -0,0 +1,52 @@
+import { renderHook } from "@testing-library/react"
+import "@testing-library/jest-dom"
+import { DivIcon } from "leaflet"
+import React from "react"
+import { MapContainer } from "react-leaflet"
+import {
+  DivIconOptions,
+  useReactDivIcon,
+} from "../../../../src/components/map/utilities/reactDivIcon"
+
+const renderHookReactDivIcon = (initialProps?: DivIconOptions) =>
+  renderHook((props) => useReactDivIcon(props), {
+    wrapper: ({ children }) => <MapContainer>{children}</MapContainer>,
+    initialProps,
+  })
+
+describe("useReactDivIcon", () => {
+  test("should return divIcon and icon element", () => {
+    const { result } = renderHookReactDivIcon()
+
+    expect(result.current.divIcon).toBeInstanceOf(DivIcon)
+    expect(result.current.iconContainer).toBeEmptyDOMElement()
+  })
+
+  test("when given custom options, should return divIcon and icon element", () => {
+    const { result } = renderHookReactDivIcon({
+      className: "testClass",
+      iconSize: [20, 20],
+    })
+
+    expect(result.current.divIcon).toBeInstanceOf(DivIcon)
+    expect(result.current.iconContainer).toBeEmptyDOMElement()
+  })
```

Comment:
**question:** is it possible to somehow assert that these options have their desired effect? Or do you think that would get too into the weeds of testing Leaflet's `divIcon` itself?

Reply:
I had to think on this for a while, I didn't think that `createIcon`[^1] was in our `@types/leaflet` file but upon checking I did find it there. The actual icon isn't created until a `Marker` calls `createIcon` so we're having to simulate this being consumed by something to assert that it's actually taking on new options.
See 4d8c219bdc99449e11de9a7dc7133fead1e4f552.

Because `Marker` also passes in the previous `div` when updating, I've also tried to simulate that in c6f2aea995c4b4c9d394730aa829f65d971340ea, but maybe it's too specific to leaflet internals and I should drop that or move it into it's own test?

[^1]: `createIcon` is in their public documentation, so it seems to technically be in the public api?

Reply:
Hmm, okay... I'm not sure I'm following 100% since this logic is all rather complicated but I think this is good enough as far as testing goes. 👍